src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.SpeedOpts.cs

```diff
@@ -0,0 +1,630 @@
+                    if (lengthToExamine >= (nuint)Vector128<byte>.Count)
+                    {
+                        Vector256<byte> values0 = Vector256.Create(value0);
+                        Vector256<byte> values1 = Vector256.Create(value1);
+                        Vector256<sbyte> valuesLessThan = Vector256.Create((sbyte)(uLessThan - 0x80));
+
+                        // Subtract Vector128<byte>.Count so we have now subtracted Vector256<byte>.Count
+                        lengthToExamine -= (nuint)Vector128<byte>.Count;
+                        // First time this checks again against 0, however we will move into final compare if it fails.
+                        while (lengthToExamine > offset)
+                        {
+                            Vector256<byte> search = LoadVector256(ref searchSpace, offset);
+                            // Bitwise Or to combine the flagged matches for the second value to our match flags
+                            matches = Avx2.MoveMask(
+                                            Avx2.Or(
+                                                Avx2.CompareGreaterThan(
+                                                    valuesLessThan,
+                                                    Avx2.Subtract(search, Vector256.Create((byte)0x80)).AsSByte()).AsByte(),
+                                                Avx2.Or(
+                                                    Avx2.CompareEqual(values0, search),
+                                                    Avx2.CompareEqual(values1, search))
+                                                )
+                                            );
+                            // Note that MoveMask has converted the equal vector elements into a set of bit flags,
+                            // So the bit position in 'matches' corresponds to the element offset.
+                            if (matches == 0)
+                            {
+                                // None matched
+                                offset += (nuint)Vector256<byte>.Count;
+                                continue;
+                            }
+
+                            goto IntrinsicsMatch;
+                        }
+
+                        {
+                            // Move to Vector length from end for final compare
+                            Vector256<byte> search = LoadVector256(ref searchSpace, lengthToExamine);
+                            offset = lengthToExamine;
+                            // Same as method as above
+                            matches = Avx2.MoveMask(
+                                            Avx2.Or(
+                                                Avx2.CompareGreaterThan(
+                                                    valuesLessThan,
+                                                    Avx2.Subtract(search, Vector256.Create((byte)0x80)).AsSByte()).AsByte(),
```

Comment:
Also `CompareGreaterThan` is only `sbyte`, so subtract `0x80` before doing the compare to keep the order correct